meson.build

```diff
@@ -7,13 +7,17 @@ wraps = get_option('wraps')
 
 foreach w : wraps
   sub = subproject(w)
-  if wraps.length() == 1
-    foreach d: get_option('depvars')
-      v = sub.get_variable(d)
-    endforeach
-  endif
 endforeach
 
+foreach name : get_option('depnames')
+  dep = dependency(name)
+  assert(dep.type_name() == 'internal')
+endforeach
+
+foreach name : get_option('prognames')
+  prog = find_program(name)
+  assert(prog.full_path().startswith(meson.global_build_root()))
+endforeach
```

Comment:
This might be a `find_program('foo.py')` instead of an `executable()`

Reply:
It can be in source dir indeed, fixed.